### PR TITLE
[move-prover] Package integration and dependency cleanup

### PR DESCRIPTION
**Pull request by wrwg**
This PR refines the integration of the Move prover into the CLI/package system as well as the Rust test infrastructure. As a result, dependencies from the move-prover crate into other crates outside of the
language tree are also eliminated.

The user level documentation is probably the best entrypoint into this work. See [here for the updated user guide](https://github.com/wrwg/diem/blob/package/language/move-prover/doc/user/prover-guide.md). It describes how to call and configure the prover via the `package` command, and how to setup tests.

One design decision here is to let the package system always verify by default a package as a whole. This is generally faster if the intent is to actually verify all modules in the package, as when running verification as a Rust test. When working from the command line one can, however, filter prover targets, for example as in `move package prover -t DiemConfig`. The string after the `-t` works as in cargo test.

The diem framework and move-stdlib have been updated to use the new prover test infra via the CLI. The prover unit tests, however, continue to use a customized test approach, because they have different requirements (exp files etc).

This PR introduces breaking changes for Move prover users:

- After this PR, anybody calling the prover outside of the move-prover crate should go via the `move package prove` command, as documented in the user guide.
- If one calls the prover directly, **it will not longer** pick an environment variable `MOVE_PROVER_CONFIG`. If you need to do this, use `--config $MOVE_PROVER_CONFIG` explicitly.  (But one really only needs to do this for running prover unit tests in the move-prover tree; see below)

For prover developers, the following changes:

- Running cargo test in `move-prover` does not longer test diem-framework or move-stdlib. Use `tools/check_pr.sh -m` to run all relevant Move verification tests.
- Tests in `move-prover/tests/sources` cannot depend longer on anything else than the Move stdlib.
- Checkout the updated [tests/README.md](https://github.com/wrwg/diem/blob/package/language/move-prover/tests/README.md) for running the prover on sources in the test tree.

The PR contains a lot of .exp changes in the move-prover/tests tree.  This is because the console output has changed, because Diem dependencies have been removed, and the .cvc4_exp files where out of date wrt from a PR last week (
they are not run in CI).

## Motivation

Package integration,

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Existing tests refactored

## Related PRs

NA

### CODE REVIEW

**Comment by wrwg** [comment]
@vgao1996

**Review by mengxu-fb — DISMISSED** [review]

**Review by mengxu-fb — DISMISSED** [review]

File: language/move-prover/doc/user/prover-guide.md [hunk 1]
```diff
@@ -1,59 +1,144 @@
...
+> that the Rust code part of the prover (and the underlying Move compiler) is by an order of magnitude faster
+> in release mode than in debug mode.
 
-In the sequel, we assume that `mvp` invokes the binary of the prover, either via an alias as above, or by
-other means.
+We assume in the sequel that the Move CLI is reachable from the command line via the `move` command
+(defined by an alias as above or by other means).
 
-## Command Line Interface
+In order to call the CLI, you must have a [*move package*](TODO). In the simplest case, a Move package is defined by a
```
Comment (tzakian) [inline language/move-prover/doc/user/prover-guide.md:32]: Nit: you can probably point this ref to here: https://diem.github.io/move/packages.html

**Review by tzakian — APPROVED** [review]
This is awesome! Super excited by this.

I left a small nit for a link to Move packages in the README but otherwise looks great to me

File: language/move-prover/doc/user/prover-guide.md (hunk 1 above)
Comment (wrwg) [reply language/move-prover/doc/user/prover-guide.md]: Done

**Comment by wrwg** [comment]
Need reapproval @mengxu-fb @tzakian

**Review by mengxu-fb — DISMISSED** [review]

**Comment by vgao1996** [comment]
A bit late to the party, but I'm super excited this is happening and thanks to @wrwg for implementing this! This is immensely helpful for getting us ready to move to the new repo!

**Pull request merged**